src/gmt_io.c

```diff
@@ -4667,12 +4667,13 @@ GMT_LOCAL void gmtio_check_abstime_format (struct GMT_CTRL *GMT, struct GMT_DATA
 		if (GMT->current.io.col_type[GMT_OUT][col] != GMT_IS_ABSTIME) continue;	/* Not an abstime column */
 		for (row = 0; row < MIN (S->n_rows, 20); row++) {	/* Maximum 20 rows are examined */
 			sub = S->data[col][row] - floor (S->data[col][row]);	/* Fractional second */
+            if (sub > sub_max) sub_max = sub;
 			if (gmt_M_is_zero (sub)) continue;	/* Not checking zeros for width */
 			if ((this_w = gmtio_get_precision_width (GMT, sub)) > w_max)
 				w_max = this_w;
 		}
 	}
-	if (w_max) {	/* Need to append w_max "x" to the default format */
+	if (w_max && (sub_max >= 1E-6)) {	/* Need to append w_max "x" to the default format */
```

Comment:
@PaulWessel, can you provide your opinion on whether this is added criteria is a good idea? The benefit is that for most cases like ```gmt math -T``` this will prevent precision being added to FORMAT_CLOCK_OUT when the maximum error is less than the least significant digit. The risk is that we're only testing for the first 20 rows, so if the error systematically grows over the course of the record it could be problematic.

Reply:
I think this is a good check. The 20 is a bit arbitrary anyway and we cannot know what lies down the road...